### DISCUSSION TITLE
Is there a way to read null records?

### DISCUSSION THREAD

**Discussion by fquinsol** [JoshClose/CsvHelper#1795] [Q&A] [answered]
CSVHelper if perfectly happy writing null records as - ,,,,,, - 
However reading back is a problem.
The might be a way to do it, but I have not figure out how.
Any suggestions are appreciated.

Thanks

**Comment by JoshClose** [comment, marked as answer]
If the type is nullable, this works without any config. If the type is a string, it'll be a string and you need to set an empty string as a null value.

```cs
void Main()
{
    var s = new StringBuilder();
    s.Append("Id,Name\r\n");
    s.Append(",\r\n");
    var config = new CsvConfiguration(CultureInfo.InvariantCulture)
    {
    };
    using (var reader = new StringReader(s.ToString()))
    using (var csv = new CsvReader(reader, config))
    {
        csv.Context.RegisterClassMap<FooMap>();
        csv.GetRecords<Foo>().ToList().Dump();
    }
}

private class Foo
{
    public int? Id { get; set; }
    public string Name { get; set; }
}

private class FooMap : ClassMap<Foo>
{
    public FooMap()
    {
        Map(m => m.Id);
        Map(m => m.Name).TypeConverterOption.NullValues(string.Empty);
    }
}
```

![image](https://user-images.githubusercontent.com/114971/117312768-efdd4200-ae4a-11eb-8194-cc0c8ff92d7e.png)

**Reply by HunterAhlquist** [reply]
I don't think it should work like that in my opinion. If a string property is nullable, it shouldn't replace null values with an empty string.
